statistics/table.go

```diff
@@ -113,51 +136,56 @@ func (h *Handle) TableStatsFromStorage(ctx context.Context, info *model.TableInf
 	for _, row := range rows {
 		distinct := row.Data[3].GetInt64()
 		histID := row.Data[2].GetInt64()
+		histVer := row.Data[4].GetUint64()
 		if row.Data[1].GetInt64() > 0 {
 			// process index
-			var idx *Index
-			for _, idxInfo := range info.Indices {
+			idx := table.Indices[histID]
+			for _, idxInfo := range tableInfo.Indices {
 				if histID == idxInfo.ID {
-					hg, err1 := histogramFromStorage(ctx, info.ID, histID, nil, distinct, 1)
-					if err1 != nil {
-						return nil, errors.Trace(err1)
+					if idx == nil || idx.LastUpdateVersion < histVer {
+						hg, err := histogramFromStorage(h.ctx, tableInfo.ID, histID, nil, distinct, 1, histVer)
+						if err != nil {
+							return nil, errors.Trace(err)
+						}
+						idx = &Index{Histogram: *hg}
 					}
-					idx = &Index{Histogram: *hg}
 					break
 				}
 			}
 			if idx != nil {
-				table.Indices[idx.ID] = idx
+				table.Indices[histID] = idx
 				indexCount++
 			} else {
-				log.Warnf("We cannot find index id %d in table %s now. It may be deleted.", histID, info.Name)
+				// TODO: If we support create index ddl, it may be a new created index. We needn't refer tableInfo any more.
+				log.Warnf("We cannot find index id %d in table %s now. It may be deleted.", histID, tableInfo.Name)
```

Comment:
When we adding a new index, this always happen, do we really need to log at warn level?